test/tests/functional/pbs_basil_support.py

```diff
@@ -362,5 +407,393 @@ def test_hbmemm_rsc(self):
+
+        cmd = ['xtprocadmin', '-k', 'm', 'batch', '-n', "%s" % (knl2)]
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
+        self.assertEqual(ret['rc'], 0)
+
+        # Wait for 12 seconds before checking
+        time.sleep(12)
+
+        # Check that the nodes are Free.
+        self.server.expect(VNODE, {'state': 'free'}, id=k1)
+        self.server.expect(VNODE, {'state': 'free'}, id=k2)
+
+        # Submit few jobs
+        a = {'Resource_List.select': '1:vntype=cray_compute'}
+        job = Job(TEST_USER, attrs=a)
+
+        job_id = self.server.submit(job)
+        self.server.expect(JOB, {'job_state': 'R'}, id=job_id)
+        # Check that exec_vnode is a KNL vnode.
+        self.server.status(JOB, 'exec_vnode', id=job_id)
+        evnode = job.execvnode()[0].keys()[0]
+        if evnode in klist:
+            self.logger.info("exec_vnode %s is a KNL vnode." % (evnode))
+            rv = 1
+        else:
+            self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
+            rv = 0
+        self.assertTrue(rv == 1)
+
+        job2 = Job(TEST_USER, attrs=a)
+
+        job_id2 = self.server.submit(job2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=job_id2)
+        # Check that exec_vnode is a KNL vnode.
+        self.server.status(JOB, 'exec_vnode', id=job_id2)
+        evnode = job2.execvnode()[0].keys()[0]
+        if evnode in klist:
+            self.logger.info("exec_vnode %s is a KNL vnode." % (evnode))
+            rv = 1
+        else:
+            self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
+            rv = 0
+        self.assertTrue(rv == 1)
+
+        job3 = Job(TEST_USER, attrs=a)
+
+        job_id3 = self.server.submit(job3)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=job_id3)
+
+        # Delete the Job.
+        self.server.delete(job_id, wait=True)
+
+        # Verify Job3 should start running
+        self.server.expect(JOB, {'job_state': 'R'}, id=job_id3)
+        # Check that exec_vnode is a KNL vnode.
+        self.server.status(JOB, 'exec_vnode', id=job_id3)
+        evnode = job3.execvnode()[0].keys()[0]
+        if evnode in klist:
+            self.logger.info("exec_vnode %s is a KNL vnode." % (evnode))
+            rv = 1
+        else:
+            self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
+            rv = 0
+        self.assertTrue(rv == 1)
+
+        # Restore hook freq to 300
+        self.server.manager(MGR_CMD_SET, PBS_HOOK,
+                            {'enabled': 'true', 'freq': 300},
+                            id='PBS_alps_inventory_check', expect=True)
+
+        # Change all nodes back to batch mode and restart PBS
+        cmd = ['xtprocadmin', '-k', 'm', 'batch']
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
```

Comment:
It is very important that you move this to tearDown().  If not, your nodes won't be in batch mode for subsequent tests and they will fail.